pkg/proxy/endpoints_test.go

```diff
@@ -45,28 +45,28 @@ func (proxier *FakeProxier) deleteEndpointSlice(slice *discovery.EndpointSlice)
 func TestGetLocalEndpointIPs(t *testing.T) {
 	testCases := []struct {
 		endpointsMap EndpointsMap
-		expected     map[types.NamespacedName]sets.String
+		expected     map[types.NamespacedName]sets.Set[string]
 	}{{
 		// Case[0]: nothing
 		endpointsMap: EndpointsMap{},
-		expected:     map[types.NamespacedName]sets.String{},
+		expected:     map[types.NamespacedName]sets.Set[string]{},
 	}, {
 		// Case[1]: unnamed port
 		endpointsMap: EndpointsMap{
 			makeServicePortName("ns1", "ep1", "", v1.ProtocolTCP): []Endpoint{
 				&BaseEndpointInfo{Endpoint: "1.1.1.1:11", IsLocal: false, Ready: true, Serving: true, Terminating: false},
 			},
 		},
-		expected: map[types.NamespacedName]sets.String{},
+		expected: map[types.NamespacedName]sets.Set[string]{},
 	}, {
 		// Case[2]: unnamed port local
 		endpointsMap: EndpointsMap{
 			makeServicePortName("ns1", "ep1", "", v1.ProtocolTCP): []Endpoint{
 				&BaseEndpointInfo{Endpoint: "1.1.1.1:11", IsLocal: true, Ready: true, Serving: true, Terminating: false},
 			},
 		},
-		expected: map[types.NamespacedName]sets.String{
-			{Namespace: "ns1", Name: "ep1"}: sets.NewString("1.1.1.1"),
+		expected: map[types.NamespacedName]sets.Set[string]{
+			{Namespace: "ns1", Name: "ep1"}: sets.New[string]("1.1.1.1"),
```

Comment:
FTR, you can just say `sets.New("1.1.1.1")` here. You only need to specify the type explicitly when you're constructing an _empty_ set and there's nothing for it to infer the type from.

(OTOH maybe it's better to be consistent, since there are a bunch of places in this file where we explicitly construct empty sets too.)